test/libsolidity/semanticTests/smoke_test.sol

```diff
@@ -23,3 +26,5 @@ contract C {
 // i() # Does not exist. # -> FAILURE # Reverts. #
 // x(bytes32): 0x31 -> 0x31
 // t(bool): true -> false
+// s(): hex"4200ef" -> 7
```

Comment:
I'm not totally sure anymore we are using non-confusing styles here...

I'm pretty sure I will use things like `0x20` at some point and not remember that it will left-align.

Perhaps it would be better to still interpret hex numbers as right-aligned numbers and instead have something like

`hexLeft"4200ef"` for left-aligned padded and then `hexUnpadded"4200ef` for unpadded? Or perhaps `rawHex`?

Reply:
I would say hex strings ``hex"00abcd"`` should always only be unpadded hex values of the size given by the string length - that's the only reason for using strings and consistent with solidity.

So if ``0x20`` should default to right-aligning (I can see the point for potential confusion otherwise), then I'd rather introduce ``left(0x20)`` for left-aligning them, instead of using hex strings for that.

Reply:
The test case should probably have leading and trailing zeros btw.

Reply:
Yes, I agree that the syntax became a little bit confusing. I would propose to use the following:

- `0x20` (right-aligned, padded)
- `right(0x20)` (right-aligned, padded)
- `left(0x20)` (left-aligned, padded)
- `hex"4200ef"` (unpadded)

I thought about different ways of how to that and concluded a built-in function to be a proper solution. There is no real to add `right(0x20)`, but I'd do that to make concept of alignment more complete.

Reply:
I think these should not be allowed:
- `hex"0x20"`
- `left(hex"4200ef")`

Reply:
if that's not too hard to parse, then it looks good, I would say.

Reply:
No, that shouln't be too hard. If we agree on the last proposal, we could merge this PR and I'll do the change in another one.